### PR TITLE
decompiled func_80242178_7ECE88

### PR DESCRIPTION
**Pull request by Muskellunge**

### CODE REVIEW

File: src/world/area_mac/mac_00/7EB340.c [hunk 1]
```diff
@@ -375,7 +375,17 @@ INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80241FF0_7ECD00);
 
 #include "world/common/UnkPositionFunc.inc.c"
 
-INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80242178_7ECE88);
+// INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80242178_7ECE88);
```
Comment (nanaian) [inline src/world/area_mac/mac_00/7EB340.c:378]: You can remove this comment

File: src/world/area_mac/mac_00/7EB340.c [hunk 2]
```diff
@@ -375,7 +375,17 @@ INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80241FF0_7ECD00);
...
+
+extern s32 mac_00_dojoBattleIDs[];
+extern EncounterStatus gCurrentEncounter;
```
Comment (nanaian) [inline src/world/area_mac/mac_00/7EB340.c:381]: I don't think this extern is required; gCurrentEncounter is declared by common.h/variables.h

Comment (ethteck) [inline src/world/area_mac/mac_00/7EB340.c:381]: This extern is unnecessary

File: src/world/area_mac/mac_00/7EB340.c (hunk 1 above)
Comment (ethteck) [inline src/world/area_mac/mac_00/7EB340.c:378]: You can remove this comment

File: src/world/area_mac/mac_00/7EB340.c [hunk 3]
```diff
@@ -375,7 +375,17 @@ INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80241FF0_7ECD00);
 
 #include "world/common/UnkPositionFunc.inc.c"
 
-INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80242178_7ECE88);
+// INCLUDE_ASM(s32, "world/area_mac/mac_00/7EB340", func_80242178_7ECE88);
+
+extern s32 mac_00_dojoBattleIDs[];
```
Comment (nanaian) [inline src/world/area_mac/mac_00/7EB340.c:380]: Consider moving this extern to the top of the file

File: ver/us/symbol_addrs.txt [hunk 4]
```diff
@@ -7732,7 +7732,7 @@ D_802483EC_7F30FC = 0x802483EC; // type:data rom:0x7F30FC
 D_80248F18_7F3C28 = 0x80248F18; // type:data rom:0x7F3C28
 D_80248F20_7F3C30 = 0x80248F20; // type:data rom:0x7F3C30
 D_80248F28_7F3C38 = 0x80248F28; // type:data rom:0x7F3C38
-D_8024A632_7F5342 = 0x8024A632; // type:data rom:0x7F5342
+mac_00_dojoBattleIDs = 0x8024A632; // type:data rom:0x7F5342
```
Comment (nanaian) [inline ver/us/symbol_addrs.txt:7735]: We moved this symbol 2 bytes earlier, so we need to adjust the addresses for this:
```suggestion
mac_00_dojoBattleIDs = 0x8024A630; // type:data rom:0x7F5340
```

File: src/world/area_mac/mac_00/7EB340.c (hunk 3 above)
Comment (ethteck) [inline src/world/area_mac/mac_00/7EB340.c:380]: This extern should go at the top of the c file with other externs (before the first c function)

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by ethteck — CHANGES_REQUESTED** [review]
Woohooooo

File: src/world/area_mac/mac_00/7EB340.c (hunk 1 above)
Comment (nanaian) [reply src/world/area_mac/mac_00/7EB340.c:378]: Also you'll need to run `./coverage.py`

**Comment by BowserSlug** [comment]
⚠️ This PR introduces a warning:
- include/variables.h:90: warning: previous declaration of `gCurrentEncounter'
- src/world/area_mac/mac_00/7EB340.c:381: warning: redundant redeclaration of `gCurrentEncounter' in same scope
- src/world/area_mac/mac_00/7EB340.c:394: warning: implicit declaration of function `sync_status_menu'
- src/world/area_mac/mac_00/7EB340.c:398: warning: function declaration isn't a prototype
- src/world/area_mac/mac_00/7EB340.c:400: warning: function declaration isn't a prototype

**Review by nanaian — APPROVED** [review]

**Pull request merged**